aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt

```diff
@@ -1,63 +1,92 @@
 package org.wordpress.aztec.spans
 
 import android.content.Context
+import android.graphics.Canvas
 import android.graphics.Paint
 import android.graphics.Rect
 import android.graphics.drawable.Drawable
+import android.text.TextUtils
+import android.text.style.DynamicDrawableSpan
 import android.text.style.ImageSpan
 import android.text.style.ParagraphStyle
 import android.view.View
 import android.widget.Toast
-
 import org.wordpress.android.util.DisplayUtils
 
-class AztecMediaSpan(val context: Context, drawable: Drawable, source: String) : ImageSpan(drawable), ParagraphStyle {
-    private val html = source
+class AztecMediaSpan @JvmOverloads constructor(val context: Context?, private var image: Drawable?, val source: String, attributes: String = "") : DynamicDrawableSpan()  {
+
+    private val TAG: String = "img"
+
+    var attributes: String = ""
 
     companion object {
         private val rect: Rect = Rect()
     }
 
+    init {
+        if (attributes.isEmpty()) {
+            this.attributes = " src=\"$source\""
+        } else {
+            this.attributes = attributes
+        }
+
+        setBounds(image)
+    }
+
+    override fun getDrawable(): Drawable? {
+        return image
+    }
+
+    fun setDrawable(newDrawable: Drawable?) {
+        image = newDrawable
+        setBounds(image)
+    }
+
     override fun getSize(paint: Paint?, text: CharSequence?, start: Int, end: Int, metrics: Paint.FontMetricsInt?): Int {
-        val drawable = drawable
-        val bounds = getBounds(drawable)
+        val drawable = image
+        val bounds = drawable?.bounds
 
-        if (metrics != null) {
+        if (metrics != null && bounds != null) {
             metrics.ascent = -bounds.bottom
             metrics.descent = 0
 
             metrics.top = metrics.ascent
             metrics.bottom = 0
         }
 
-        return bounds.right
+        return bounds?.right ?: 0
     }
 
-    private fun getBounds(drawable: Drawable): Rect {
-        if (drawable.intrinsicWidth === 0) {
-            rect.set(0, 0, drawable.intrinsicWidth, drawable.intrinsicHeight)
-            return rect
-        }
+    override fun draw(canvas: Canvas, text: CharSequence, start: Int, end: Int, x: Float,
+                      top: Int, y: Int, bottom: Int, paint: Paint) {
+        val b = image
+        canvas.save()
 
-        /*
-         * Following Android guidelines for keylines and spacing, screen edge margins should
-         * be 16dp.  Therefore, the width of images should be the width of the screen minus
-         * 16dp on both sides (i.e. 16 * 2 = 32).
-         *
-         * https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-baseline-grids
-         */
```

Comment:
These comments should be moved [here](https://github.com/wordpress-mobile/WordPress-Aztec-Android/pull/202/files#diff-d6152856756259d707419e470fe4f340R79) in the `setBounds` function to clarify how the width is determined.